test/integration/visitor_redirected_to_login_when_clicks_checkout_test.rb

```diff
@@ -0,0 +1,26 @@
+require 'test_helper'
+
+class VisitorRedirectedToLoginWhenClicksCheckoutTest < ActionDispatch::IntegrationTest
+  test 'user that has not logged in directed to login when click checkout' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.